tests/vmi_networking_test.go

```diff
@@ -629,70 +620,125 @@ var _ = Describe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com][le
-				} else {
-					if ipv4NetworkCIDR == "" {
-						ipv4NetworkCIDR = api.DefaultVMCIDR
-					}
-					By("Checking ping (IPv4) to gateway")
-					ipAddr := gatewayIPFromCIDR(ipv4NetworkCIDR)
-					Expect(libnet.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
+			table.DescribeTable("ipv4", func(ports []v1.Port, withCustomCIDR bool) {
+				var clientVMI *v1.VirtualMachineInstance
+				var networkCIDR string
 
-					By("Checking ping (IPv4) to google")
-					Expect(libnet.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
-					Expect(libnet.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
+				if withCustomCIDR {
+					networkCIDR = "10.10.10.0/24"
 				}
 
-				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed())
+				// Create the client only one time
+				if clientVMI == nil {
+					clientVMI = masqueradeVMI([]v1.Port{}, networkCIDR)
+					_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(clientVMI)
+					Expect(err).ToNot(HaveOccurred())
+					clientVMI = tests.WaitUntilVMIReady(clientVMI, console.LoginToCirros)
+				}
 
-				By("Connecting from the client vm")
-				err = console.SafeExpectBatch(clientVMI, createExpectConnectToServer(serverIP, tcpPort, true), 30)
-				Expect(err).ToNot(HaveOccurred())
+				serverVMI = masqueradeVMI(ports, networkCIDR)
 
-				By("Rejecting the connection from the client to unregistered port")
-				err = console.SafeExpectBatch(clientVMI, createExpectConnectToServer(serverIP, tcpPort+1, false), 30)
+				serverVMI.Labels = map[string]string{"expose": "server"}
+				_, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Create(serverVMI)
 				Expect(err).ToNot(HaveOccurred())
-			}
+				serverVMI = tests.WaitUntilVMIReady(serverVMI, console.LoginToCirros)
+				Expect(serverVMI.Status.Interfaces).To(HaveLen(1))
+				Expect(serverVMI.Status.Interfaces[0].IPs).NotTo(BeEmpty())
 
-			err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Delete(serverVMI.Name, &v13.DeleteOptions{})
-			Expect(err).ToNot(HaveOccurred())
-		}, table.Entry("with a specific port number", []v1.Port{{Name: "http", Port: 8080}}, defaultCIDR),
-			table.Entry("without a specific port number", []v1.Port{}, defaultCIDR),
-			table.Entry("with custom CIDR", []v1.Port{}, customCIDR),
-		)
+				By("starting a tcp server")
+				tcpPort := 8080
+				tests.StartTCPServer(serverVMI, tcpPort)
+
+				if networkCIDR == "" {
+					networkCIDR = api.DefaultVMCIDR
+				}
+
+				By("Checking ping (IPv4) to gateway")
+				ipAddr := gatewayIPFromCIDR(networkCIDR)
+				Expect(libnet.PingFromVMConsole(serverVMI, ipAddr)).To(Succeed())
+
+				By("Checking ping (IPv4) to google")
+				Expect(libnet.PingFromVMConsole(serverVMI, "8.8.8.8")).To(Succeed())
+				Expect(libnet.PingFromVMConsole(clientVMI, "google.com")).To(Succeed())
+
+				verifyClientServerConnectivity(clientVMI, serverVMI, tcpPort, k8sv1.IPv4Protocol)
```

Comment:
this function returns `error` - as it should.

You must `Expect(...).To(Succeed())` otherwise, nothing is tested.

Reply:
Great catch! Thank you!